tests/frontend/test_configuration.py

```diff
@@ -396,3 +396,89 @@ def test_parse_environment_variable_integer(self, monkeypatch):
 
         monkeypatch.setattr(conf, "DEFAULT_CONFIG_SPEC", {"api": {"some_integer": (int, 123)}})
         assert conf.parse_environment_variable("some_integer", "123") == 123
+
+DEFAULT_KWARGS = {
+        "api": {
+            "authentication_token": "071cdcce-9241-4965-93af-4a4dbc739135",
+            "hostname": "localhost",
+            "use_ssl": True,
+            "port": 443,
+                }
+        }
+
+class TestStoreAccount:
+    """Tests for the store_account function."""
+
+    def test_config_created_locally(self, monkeypatch, tmpdir):
+        """Tests that a configuration file was created in the current
+        directory."""
+
+        test_filename = "test_config.toml"
+
+        call_history = []
+        with monkeypatch.context() as m:
+            m.setattr(os, "getcwd", lambda: tmpdir)
+            m.setattr(conf, "user_config_dir", lambda *args: "NotTheCorrectDir")
+            m.setattr(conf, "save_config_to_file", lambda a, b: call_history.append((a, b)))
+            conf.store_account(authentication_token, filename=test_filename, location="local", **DEFAULT_KWARGS)
+
+        assert call_history[0][0] == EXPECTED_CONFIG
```

Comment:
Hmm, it doesn't necessarily have to be a list, though I'm actually not too sure how we could have a one-element solution here with `lambda` that mocks the line of execution, but mutates a variable in the test :thinking:

Reply:
Yep, so after giving it another round of thought, with this flow of logic we would like to use some mutable built-ins (and not `unittest.mock MagicMock` for example). But I'm up for changing it to something more fitting :)

Reply:
Is it sufficient to simply check that the file `tmpdir/config.toml` exists? It might not be as 'pure' as the above approach, but might be cleaner(?)

Reply:
I've added a solution that:
* uses a new class used for mocking
* mocks out `create_config` as well

This is because this unit test would like to merely check, that the functions created in `configuration.py` are called with the correct arguments (as per the logic of `store_account`). This way, we are not depending on the logic contained in those at all (hence there could be no side effects).

Further to this, I have included two integration tests, where the only thing that is mocked out, is either `os.getcwd` or `user_config_dir`, such that the `tmpdir` fixture can be used in these tests (and no state is left after the tests).

These tests test that the functions created in `configuration.py` indeed integrate correctly when `store_account` calls on them.

Reply:
@antalszava which commit added the solution/tests you mention above? I can't seem to find them

Reply:
Sorry, I had this commit locally at this time, now it is indeed changed!